### PR TITLE
[#576] test new admin mode in atomic metadata API endpoint

### PR DESCRIPTION
**Pull request by d-w-moore**
Test for atomically setting metadata using the admin flag.

### CODE REVIEW

File: irods/test/meta_test.py [hunk 1]
```diff
@@ -338,6 +338,32 @@ def test_metadata_manipulations_with_admin_kw__364__365(self):
...
+            # Log in as rodsuser and create a data object owned by that user.
+            ses = iRODSSession (port = adm.port,zone = adm.zone,host = adm.host, user = user.name, password = 'bpass')
+            home = helpers.home_collection(ses)
+            d = ses.data_objects.create('{home}/issue_576'.format(**locals()))
+            d_path = d.path
+
+            # Do and test the results of the atomic set using the admin session, with the ADMIN_KW turned on.
+            da = adm.data_objects.get(d_path)
+            avu_ = iRODSMeta('issue_576','dummy_value')
+            da.metadata(admin=True).apply_atomic_operations(AVUOperation(operation = "add", avu = avu_))
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/test/meta_test.py:358]: probably wouldn't conflict to name it `avu`, as it's a different namespace.

File: irods/test/meta_test.py [hunk 2]
```diff
@@ -338,6 +338,32 @@ def test_metadata_manipulations_with_admin_kw__364__365(self):
...
+            adm = self.sess
+            # Create a rodsuser
+            user = adm.users.create ( 'bobby','rodsuser' )
+            user.modify('password','bpass')
+
+            # Log in as rodsuser and create a data object owned by that user.
+            ses = iRODSSession (port = adm.port,zone = adm.zone,host = adm.host, user = user.name, password = 'bpass')
+            home = helpers.home_collection(ses)
+            d = ses.data_objects.create('{home}/issue_576'.format(**locals()))
+            d_path = d.path
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/test/meta_test.py]: Indeed not, and `path` is just a plain attribute, so might as well use it.

File: irods/test/meta_test.py [hunk 3]
```diff
@@ -338,6 +338,32 @@ def test_metadata_manipulations_with_admin_kw__364__365(self):
             if d: d.unlink(force=True)
             if user: user.remove()
 
+    def test_atomic_metadata_operations_with_admin_kw__issue_576(self):
+        ses = d = user = None
+        try:
+            adm = self.sess
+            # Create a rodsuser
+            user = adm.users.create ( 'bobby','rodsuser' )
+            user.modify('password','bpass')
```
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/test/meta_test.py]: Believe I've addressed all of these.

File: irods/test/meta_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (d-w-moore) [reply irods/test/meta_test.py]: Going to call it `avu_item` to be clearer.

**Comment by korydraughn** [comment]
If the test passes, squash to taste, no #'s

**Review by alanking — COMMENTED** [review]
Yep, agreed. Squash em

**Comment by d-w-moore** [comment]
Also adding a check for iRODS server version, to skip test if necessary for 4.2.11 and before. Reason given in commit message.

File: irods/test/meta_test.py [hunk 4]
```diff
@@ -338,6 +338,35 @@ def test_metadata_manipulations_with_admin_kw__364__365(self):
             if d: d.unlink(force=True)
             if user: user.remove()
 
+    def test_atomic_metadata_operations_with_admin_kw__issue_576(self):
+        ses = data = user = None
+        adm = self.sess
+
+        if adm.server_version <= (4,2,11):
+            self.skipTest('ADMIN_KW not valid for Metadata API in iRODS 4.2.11 and previous')
```
Comment (alanking) [inline irods/test/meta_test.py:346]: This is true even for the atomic metadata API? I see this skip elsewhere in these tests but that's for the legacy metadata API. Did you discover this through testing?

Comment (alanking) [reply irods/test/meta_test.py:346]: Ooh, found it: https://github.com/irods/irods/issues/6198 Resolving.

**Review by alanking — COMMENTED** [review]
Yes, please squash/arrange commits to taste. No #s.

**Comment by d-w-moore** [comment]
Squash done, no #'s yet.

**Review by korydraughn — APPROVED** [review]
If the test passes, pound it.

**Review by alanking — APPROVED** [review]

**Comment by d-w-moore** [comment]
Done! Test passes, pound added.

**Pull request merged**